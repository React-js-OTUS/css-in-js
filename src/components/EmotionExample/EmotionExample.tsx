import * as React from 'react';
import { FC, useState } from 'react';
import variables from '../../variables.scss';
import styled from '@emotion/styled'
import { css } from '@emotion/react';

const common = css`
  font-size: 12px;
  color: #8B97A5;
`;

const Text = styled.div<{ opened: boolean }>`
  ${common};
  display: ${props => props.opened ? 'block' : 'none'};
`

const Root = styled.div`
  .some-global-class {
    ${common};
    color: ${variables.pink}
  }
`

const Button = styled.button`
  border-radius: 8px;
  font-size: 18px;
  padding: 8px 16px;
  outline: none;
  border: 0;
  background-color: ${variables.red};
  color: #fff;
  cursor: pointer;

  &:hover {
    background-color: ${variables.redLight};
  }

  &:active {
    background-color: ${variables.redDark}
  }
`

export type ButtonProps = React.HTMLAttributes<HTMLButtonElement> & {
  className?: string;
};

export const EmotionExample: FC<ButtonProps> = ({ className, ...props }) => {
  const [opened, setOpened] = useState<boolean>(false);
  return (
    <Root className={className}>
      <Button {...props} onClick={() => setOpened(v => !v)} />
      <Text opened={opened}>Test</Text>
      <div className="some-global-class">Global</div>
    </Root>
  );
}