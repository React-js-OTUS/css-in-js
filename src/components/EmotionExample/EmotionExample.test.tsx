import * as React from 'react';
import { render } from '@testing-library/react';
import { EmotionExample } from './EmotionExample';

describe('EmotionExample', () => {
  it('base snapshot', () => {
    const { container } = render(<EmotionExample>EmotionExample</EmotionExample>);
    expect(container).toMatchSnapshot();
  });
});
