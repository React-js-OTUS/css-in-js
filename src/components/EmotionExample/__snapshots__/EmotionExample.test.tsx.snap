// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmotionExample base snapshot 1`] = `
<div>
  <div
    class="css-jcv2sj"
  >
    <button
      class="css-1k5qnfk"
    >
      EmotionExample
    </button>
    <div
      class="css-14x91a0"
    >
      Test
    </div>
    <div
      class="some-global-class"
    >
      Global
    </div>
  </div>
</div>
`;
